pkg/btree/bptree_test.go

```diff
@@ -614,7 +614,19 @@ func TestBPTree_Iteration_Overcount(t *testing.T) {
 		buf := make([]byte, 8)
 		binary.BigEndian.PutUint64(buf, math.Float64bits(23))
 
-		if err := tree.Insert(ReferencedValue{Value: buf, DataPointer: pointer.MemoryPointer{Offset: uint64(i)}}, pointer.MemoryPointer{Offset: uint64(i), Length: uint32(len(buf))}); err != nil {
+		if err := tree.Insert(
+			ReferencedValue{
+				Value: buf,
+				DataPointer: pointer.MemoryPointer{
+					Offset: uint64(i),
+					Length: uint32(9),
```

Comment:
Some test cases failed because of one use case where we attempted to insert multiple items, but we zeroed out the MemoryPointer. This is an issue because for identical nodes, we negate `Length` to indicate there should be a copy. However, `-0` is `0` so it will attempt to read from the buffer which is not correct.

Reply:
Ahhh, to address that, store `-(length + 1)` instead. Conveniently, this is just the `~` operator, so store `~length` instead of `-length`.